diff --git a/cmake/Platform/Initialization/CompilerSettings.cmake b/cmake/Platform/Initialization/CompilerSettings.cmake
index 4dbbb9b..224063f 100644
--- a/cmake/Platform/Initialization/CompilerSettings.cmake
+++ b/cmake/Platform/Initialization/CompilerSettings.cmake
@@ -13,11 +13,11 @@ function(setup_c_flags)
     if (NOT DEFINED ARDUINO_C_FLAGS)
         set(ARDUINO_C_FLAGS "-mcall-prologues -ffunction-sections -fdata-sections")
     endif (NOT DEFINED ARDUINO_C_FLAGS)
-    set(CMAKE_C_FLAGS "-g -Os       ${ARDUINO_C_FLAGS}" CACHE STRING "")
-    set(CMAKE_C_FLAGS_DEBUG "-g           ${ARDUINO_C_FLAGS}" CACHE STRING "")
-    set(CMAKE_C_FLAGS_MINSIZEREL "-Os -DNDEBUG ${ARDUINO_C_FLAGS}" CACHE STRING "")
-    set(CMAKE_C_FLAGS_RELEASE "-Os -DNDEBUG -w ${ARDUINO_C_FLAGS}" CACHE STRING "")
-    set(CMAKE_C_FLAGS_RELWITHDEBINFO "-Os -g       -w ${ARDUINO_C_FLAGS}" CACHE STRING "")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g ${ARDUINO_C_FLAGS}" CACHE STRING "")
+    set(CMAKE_C_FLAGS_DEBUG "" CACHE STRING "")
+    set(CMAKE_C_FLAGS_MINSIZEREL "-Os -DNDEBUG" CACHE STRING "")
+    set(CMAKE_C_FLAGS_RELEASE "-Os -DNDEBUG -w" CACHE STRING "")
+    set(CMAKE_C_FLAGS_RELWITHDEBINFO "-Os -w" CACHE STRING "")
 endfunction()
 
 #=============================================================================#
@@ -35,11 +35,11 @@ function(setup_cxx_flags)
     if (NOT DEFINED ARDUINO_CXX_FLAGS)
         set(ARDUINO_CXX_FLAGS "${ARDUINO_C_FLAGS} -fno-exceptions")
     endif (NOT DEFINED ARDUINO_CXX_FLAGS)
-    set(CMAKE_CXX_FLAGS "-g -Os       ${ARDUINO_CXX_FLAGS}" CACHE STRING "")
-    set(CMAKE_CXX_FLAGS_DEBUG "-g           ${ARDUINO_CXX_FLAGS}" CACHE STRING "")
-    set(CMAKE_CXX_FLAGS_MINSIZEREL "-Os -DNDEBUG ${ARDUINO_CXX_FLAGS}" CACHE STRING "")
-    set(CMAKE_CXX_FLAGS_RELEASE "-Os -DNDEBUG ${ARDUINO_CXX_FLAGS}" CACHE STRING "")
-    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-Os -g       ${ARDUINO_CXX_FLAGS}" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g ${ARDUINO_CXX_FLAGS}" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS_DEBUG "" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS_MINSIZEREL "-Os -DNDEBUG" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS_RELEASE "-Os -DNDEBUG" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-Os" CACHE STRING "")
 endfunction()
 
 #=============================================================================#
@@ -55,11 +55,8 @@ endfunction()
 #=============================================================================#
 function(setup_exe_linker_flags)
     set(ARDUINO_LINKER_FLAGS "-Wl,--gc-sections -lm")
-    set(CMAKE_EXE_LINKER_FLAGS "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_EXE_LINKER_FLAGS_DEBUG "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_EXE_LINKER_FLAGS_MINSIZEREL "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_EXE_LINKER_FLAGS_RELEASE "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
+    set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} ${ARDUINO_LINKER_FLAGS}")
 endfunction()
 
 #=============================================================================#
@@ -74,17 +71,8 @@ endfunction()
 # Setups some basic flags for the gcc linker to use when linking shared libraries.
 #=============================================================================#
 function(setup_shared_lib_flags)
-    set(CMAKE_SHARED_LINKER_FLAGS "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_SHARED_LINKER_FLAGS_DEBUG "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-
-    set(CMAKE_MODULE_LINKER_FLAGS "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_MODULE_LINKER_FLAGS_DEBUG "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_MODULE_LINKER_FLAGS_RELEASE "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
-    set(CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO "${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
+    set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${ARDUINO_LINKER_FLAGS}" CACHE STRING "")
 endfunction()
 
 #=============================================================================#
